explorations/exploring_recipes.Rmd

```diff
@@ -0,0 +1,265 @@
+  "metformin", "repaglinide", "nateglinide", "chlorpropamide", "glimepiride", 
+  "acetohexamide", "glipizide","glyburide","tolbutamide","pioglitazone","rosiglitazone",
+  "acarbose","miglitol","troglitazone","tolazamide","examide","citoglipton","insulin",
+  "glyburide.metformin","glipizide.metformin","glimepiride.pioglitazone","metformin.rosiglitazone",
+  "metformin.pioglitazone","change","diabetesMed", "readmitted")
+df[cols_to_factor] <- lapply(df[cols_to_factor], as.factor)
+
+df$age <- factor(df$age, ordered = TRUE)
+
+df[c("weight", "max_glu_serum", "A1Cresult")] <- lapply(
+  df[c("weight", "max_glu_serum", "A1Cresult")], as.numeric)
+
+data <- df %>% slice(1:20000)
+
+glimpse(data)
+
+print("size of dataset:")
+print(dim(data))
+
+print("unique patients:")
+print(length(unique(data$patient_nbr)))
+
+print("missing percentage:")
+print(countMissingData(data))
+
+print("structure:")
+print(str(data))
+```
+
+Train test split
+Splitting on the readmitted flag makes sure that the distribution is the same in both sets.
+```{r}
+train_index <- createDataPartition(data$readmitted,
+  p = 0.8,
+  times = 1,
+  list = TRUE)
+
+data_train <- data[train_index$Resample1, ]
+data_test <- data[-train_index$Resample1, ]
+```
+
+
+### Basic Recipe
+Use `?selections` to see how to select certain variables. One can select by type, role, or name
+This one does centering and scaling of numeric variables.
+```{r}
+# Initialize
+rec_obj <- recipe(readmitted ~ ., data = data)
+
+# Create recipe
+standardized <- rec_obj %>%
+  step_center(all_numeric()) %>%
+  step_scale(all_numeric()) 
+standardized
+
+# Train recipe
+trained_rec <- prep(standardized, training = data_train)
+
+# Apply recipe
+data_train <- bake(trained_rec, newdata = data_train)
+data_test  <- bake(trained_rec, newdata = data_test)
+```
+
+Add more steps with by just adding to the trained recipe. Get a list of steps using `apropos("^step_")`
+```{r}
+apropos("^step_")
+```
+
+Updating a recipe
+```{r}
+# Update recipe
+trained_rec <- trained_rec %>%
+  step_nzv(all_predictors())
+
+# Train updated recipe (recall that you'll have to start with a fresh copy of
+# data_train to apply the recipe from the beginning).
+data_train <- data[train_index$Resample1, ]
+data_test <- data[-train_index$Resample1, ]
+trained_rec <- prep(trained_rec, training = data_train)
+
+# Apply recipe
+data_train <- bake(trained_rec, newdata = data_train)
+data_test  <- bake(trained_rec, newdata = data_test)
+```
+
+Dummification
+```{r}
+# Update recipe
+trained_rec <- trained_rec %>%
+  step_dummy(starts_with("gender"))
+
+# Train updated recipe
+data_train <- data[train_index$Resample1, ]
+data_test <- data[-train_index$Resample1, ]
+trained_rec <- prep(trained_rec, training = data_train)
+
+# Apply recipe
+data_train <- bake(trained_rec, newdata = data_train)
+data_test  <- bake(trained_rec, newdata = data_test)
+```
+
+## Adding a custom step
```

Comment:
This is awesome! It's not easy for me to see what the arguments to various functions are doing, if you feel like adding some comments...

Reply:
I added a lot of comments. And a print method. Hopefully it's more clear